waku/v2/protocol/content_topic.go

```diff
@@ -0,0 +1,110 @@
+package protocol
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+const DefaultContentTopic = "/waku/2/default-content/proto"
+
+var ErrInvalidFormat = errors.New("invalid format")
+var ErrMissingGeneration = errors.New("missing part: generation")
+var ErrInvalidGeneration = errors.New("generation should be a number")
+
+type ContentTopic struct {
+	*ContentTopicParams
+	ApplicationName    string
+	ApplicationVersion uint32
+	ContentTopicName   string
+	Encoding           string
+}
+
+type ContentTopicParams struct {
+	Generation int
+}
+
+type ContentTopicOption func(*ContentTopicParams)
+
+func (ct ContentTopic) String() string {
+	return fmt.Sprintf("/%s/%d/%s/%s", ct.ApplicationName, ct.ApplicationVersion, ct.ContentTopicName, ct.Encoding)
+}
+
+func NewContentTopic(applicationName string, applicationVersion uint32,
+	contentTopicName string, encoding string, opts ...ContentTopicOption) (ContentTopic, error) {
+
+	params := new(ContentTopicParams)
+	optList := DefaultOptions()
+	optList = append(optList, opts...)
+	for _, opt := range optList {
+		opt(params)
+	}
+	if params.Generation > 0 {
+		return ContentTopic{}, ErrInvalidGeneration
+	}
+	return ContentTopic{
+		ContentTopicParams: params,
+		ApplicationName:    applicationName,
+		ApplicationVersion: applicationVersion,
+		ContentTopicName:   contentTopicName,
+		Encoding:           encoding,
+	}, nil
+}
+
+func WithGeneration(generation int) ContentTopicOption {
+	return func(params *ContentTopicParams) {
+		params.Generation = generation
+	}
+}
+
+func DefaultOptions() []ContentTopicOption {
+	return []ContentTopicOption{
+		WithGeneration(0),
+	}
+}
+
+func (ct ContentTopic) Equal(ct2 ContentTopic) bool {
+	return ct.ApplicationName == ct2.ApplicationName && ct.ApplicationVersion == ct2.ApplicationVersion &&
+		ct.ContentTopicName == ct2.ContentTopicName && ct.Encoding == ct2.Encoding
+}
```

Comment:
Should the `ContentTopicParams` be taken into account in this function?

Reply:
Yup, we should include that as well. Although for now it won't make a difference. 
Will update.